routetable/route_table.go

```diff
@@ -535,100 +798,73 @@ func (r *RouteTable) syncRoutesForLink(ifaceName string) error {
 			continue
 		}
 
-		routeExpected := expectedCIDRs.Contains(dest)
+		expectedTarget, expectedTargetFound := expectedTargets[dest]
+		routeExpected := expectedTargetFound || (r.ipVersion == 6 && dest == ipV6LinkLocalCIDR)
 		var routeProblems []string
 		if !routeExpected {
 			routeProblems = append(routeProblems, "unexpected route")
 		}
 		if !r.deviceRouteSourceAddress.Equal(route.Src) {
+			//TODO(rlb): What about ipv6 link local?  Should we remove, if so, how does it get added back?
```

Comment:
We've always let the kernel handle the link-locals before, maybe we should just ignore those completely?  The source will be our own link local, I'd have thought.

Reply:
As discussed, I now think this is source IP selection tweaking and I think it should be inside the following conditional.  It probably works in the mainline case because  `deviceRouteSourceAddress` is empty and so is `Src` for the IPv6 route (but I haven't verified that).